chiselFrontend/src/main/scala/chisel3/Bits.scala

```diff
@@ -1830,216 +1807,193 @@ package experimental {
-      val other = that
-      requireIsHardware(this, s"'this' ($this)")
-      requireIsHardware(other, s"'other' ($other)")
-      //    pushOp(DefPrim(sourceInfo, dest, SqueezeOp, this.ref, other.ref, ILit(2)))
-      pushOp(DefPrim(sourceInfo, dest, SqueezeOp, this.ref, other.ref))
-    }
-
     /**
-      * Wrap this interval into the range determined by that UInt
+      * Squeeze this interval into the range determined by that UInt
       * Currently, that must have a defined width
       * @param that an UInt whose properties determine the squeezing
       * @return
       */
-    final def wrap(that: UInt): Interval = macro SourceInfoTransform.thatArg
-    def do_wrap(that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
-      //binop(sourceInfo, TypePropagate(_root_.firrtl.PrimOps.Wrap, Seq(this, that), Nil), WrapOp, that)
+    final def squeeze(that: UInt): Interval = macro SourceInfoTransform.thatArg
+    def do_squeeze(that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
+      //binop(sourceInfo, TypePropagate(_root_.firrtl.PrimOps.Wrap, Seq(this, that), Nil), SqueezeOp, that)
       that.widthOption match {
         case Some(w) =>
           val u = BigDecimal(BigInt(1) << w) - 1
-          do_wrap(0.U.asInterval(IntervalRange(firrtlir.Closed(0), firrtlir.Closed(u), BinaryPoint(0))))
+          do_squeeze(0.U.asInterval(IntervalRange(firrtlir.Closed(0), firrtlir.Closed(u), BinaryPoint(0))))
         case _ =>
-          throwException("wrap requires an UInt argument with a known width")
+          throwException("squeeze requires an UInt argument with a known width")
       }
     }
 
     /**
-      * Wrap this interval into the range determined by an SInt
+      * Squeeze this interval into the range determined by an SInt
       * Currently, that must have a defined width
       * @param that an SInt whose properties determine the squeezing
       * @return
       */
-    final def wrap(that: SInt): Interval = macro SourceInfoTransform.thatArg
-    def do_wrap(that: SInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
-      //binop(sourceInfo, TypePropagate(_root_.firrtl.PrimOps.Wrap, Seq(this, that), Nil), WrapOp, that)
+    final def squeeze(that: SInt): Interval = macro SourceInfoTransform.thatArg
+    def do_squeeze(that: SInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
+      //binop(sourceInfo, TypePropagate(_root_.firrtl.PrimOps.Wrap, Seq(this, that), Nil), SqueezeOp, that)
       that.widthOption match {
         case Some(w) =>
           val l = -BigDecimal(BigInt(1) << (that.getWidth - 1))
           val u = BigDecimal(BigInt(1) << (that.getWidth - 1)) - 1
-          do_wrap(Wire(Interval(IntervalRange(firrtlir.Closed(l), firrtlir.Closed(u), BinaryPoint(0)))))
+          do_squeeze(Wire(Interval(IntervalRange(firrtlir.Closed(l), firrtlir.Closed(u), BinaryPoint(0)))))
```

Comment:
This isn't right; squeeze returns the intersection of the ranges. It should be u if u is smaller than this.range.max (same for min, where its l or this.range.min, whichever is larger).